hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,133 @@
+#include <moveit_msgs/DisplayTrajectory.h>
+
+namespace Hamilton
+{
+  using namespace descartes_core;
+  using namespace descartes_trajectory;
+
+  /* ===== utilities ===== */
+  typedef std::vector<descartes_core::TrajectoryPtPtr> TrajectoryVec;
+  typedef TrajectoryVec::const_iterator TrajectoryIter; //TODO redundant use auto. 
+  enum TrajectorySegmentType {FREE_MOTION, PROCESS_PATH, FREE_MOTION_START_END_ONLY};
+  
+  /*  =============================== Hybrid Configuration Struct ===============================
+  * Is a member of HybridPlanner class and allows to define a few parameters required by MoveIt! and Descartes*/
+
+  struct HybridConfiguration
+  {
+    std::string group_name;        // = "manipulator";               /* Name of the manipulation group containing the relevant links in the robot, which is what you define when running MoveitSetupAssistant */
+    std::string robot_description; // = "robot_description";         /* Name of description on parameter server. Typically just "robot_description". */
+    std::string tcp_frame;         // = "tool0";                     /* Tool center point frame (name of link associated with tool. Usually the last link in the kinematic chain of the robot */
+    std::string base_link;         //                                /* The name of the base link of the robot */
+    std::string world_frame;       // = "base_link" OR "world_frame";/* The name of the world link in the URDF file, the frame in which you are expressing poses. */
+                                                                     /* Typically "world_frame" or "base_link". */
+    std::vector<std::string> joint_names;                            /* A list with the names of the mobile joints in the robot */ 
+                                                                     // 2nd param of toROSJointTrajectory()
+    // MoveIt! params
+    double plan_time;   // = 10.0;                      
+    double step_size;  // = 0.01; 
+    double jump_thresh; // =0;
+    bool moveit_replan;
+    bool avoid_collision;
+    std::string moveit_planner_type;
+    
+    // Descartes params
+    std::string descartes_planner_type;
+    double time_delay; // = 1.0; Last param of toROSJointTrajectory() 
+    double time_offset; // = 0.0; Used inside toROSJointTrajectory()
+  };
+
+  /*  =============================== TrajectorySegment Struct ===============================
+  * Struct to store the unplanned segments, and categorize them as free and process path. 
+  * Note: The third type "FREE_MOTION_START_END_ONLY" isn't used right now.
+  * The idea is to call the MoveIt planner via a MotionPlanning Request in this case, via the PlanHybrid method.  
+  * TODO Add more data structures in TrajectorySegment - joint vals, descartes type itself, etc */
+
+  struct TrajectorySegment
+  {
+    TrajectorySegmentType type_;      
+    std::vector<geometry_msgs::Pose> waypoints_geom_msgs_;
+  }; 
```

Comment:
See google style guide for naming convention of struct members [here](https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Variable_Names)